src/libraries/System.Diagnostics.Process/tests/ProcessTests.cs

```diff
@@ -2378,6 +2378,145 @@ public void Start_PassesArgumentsList_WhichGetsEscaped()
             }
         }
 
+        [Theory]
+        [InlineData(true, true)]
+        [InlineData(true, false)]
+        [InlineData(false, true)]
+        [InlineData(false, false)]
+        public async Task CancelRead_WaitForExit_ReturnsImmediately(bool cancelByKill, bool useAsyncAPI)
+        {
+            RemoteInvokeOptions options = new RemoteInvokeOptions();
+            options.CheckExitCode = false;
+            options.StartInfo.RedirectStandardOutput = true;
+            options.StartInfo.RedirectStandardError = true;
+            using var handle = RemoteExecutor.Invoke(() =>
+            {
+                try
+                {
+                    // Start a child that will keep the standard streams open longer than its parent.
+                    var process = Process.Start("sleep", "600"); // sleep 10 minutes.
+
+                    // Signal the test process.
+                    Console.WriteLine("Sleep child started");
+                }
+                catch (Exception e)
+                {
+                    Console.WriteLine(e.Message.Replace(Environment.NewLine, "  "));
+                }
+            }, options);
+
+            // Wait for the sleep child to be started.
+            Process process = handle.Process;
+            TaskCompletionSource<string> childOutputTcs = new();
+            TaskCompletionSource<object> eofTcs = new();
+            process.OutputDataReceived += (object sender, DataReceivedEventArgs e) =>
+            {
+                childOutputTcs.TrySetResult(e.Data);
+                if (e.Data == null)
+                {
+                    eofTcs.SetResult(null);
+                }
+            };
+            process.BeginOutputReadLine();
+            process.BeginErrorReadLine();
+            string childOutput = await childOutputTcs.Task.TimeoutAfter(30_000);
+            Assert.Equal("Sleep child started", childOutput);
+
+            if (cancelByKill)
+            {
+                process.Kill();
+            }
+            else
+            {
+                process.CancelOutputRead();
+                process.CancelErrorRead();
+            }
+
+            Task asyncWaitTask = useAsyncAPI ? process.WaitForExitAsync() :
+                                               Task.Run(() => process.WaitForExit());
+            await asyncWaitTask.TimeoutAfter(30_000);
```

Comment:
The `sleep` child is meant to quit after 10 minutes. Is there a point in waiting 30 seconds? Seems like a lot.